### ISSUE TITLE
Some questions about training

### ISSUE THREAD

**Issue by linxiang4200** [tencent-ailab/IP-Adapter#100]
Hello, thank you very much for sharing your great work！ 
I have a few questions that I would like to ask:

    1.During training, the paper describes image-text pairs, but the training code uses the same image as both input and target, and it mentions that text can be omitted. So, during training, what is the target for the training? What is the purpose of including text descriptions?

    2.Are there specific requirements for the text descriptions in the training dataset? Is it sufficient to provide simple image descriptions, or is it better to have more detailed descriptions?

    3.What are the specific differences between your approach and stable diffusion's image-to-image generation?

My questions may be a bit extensive, but I hope to receive your response. 
Once again, thank you for your outstanding work. Thank you.

**Comment by xiaohu2015** [comment]
@linxiang4200 
1) in the training, we just use the image-text pairs (a image and its caption) which are used to train text-to-image models. we want to train a text compatible image prompt adapter, so we also use text condition. As we freeze the UNet (the UNet don't lost its text prompt ability), so we think we also can use only image condition in the training.
2) we don't handle the text descriptions specially. But we think it will be better if use detailed descriptions. As images tend to contain more information than text, hence IP-Adapter is often overfitted to image condition, but we can use a lower scale to avoid this problem.
3) The SD image-to-image generation is good at maintaining structural consistency, but it can't capture semantic information especially when the strength is relatively high.

**Comment by linxiang4200** [comment]
@xiaohu2015 
thanks for your reply.
I still have a bit of confusion about the first question. If the training image and the target image are the same, what is the model learning during the training process?

**Comment by xiaohu2015** [comment]
> @xiaohu2015 thanks for your reply. I still have a bit of confusion about the first question. If the training image and the target image are the same, what is the model learning during the training process?

good question.

For IP-Adapter, we use only global image embedding of CLIP image encoder (e.g. 1024 tensor for ViT-H), hence it only capture semantic information of the reference image, but can't reconstruct the original image, hence it learns to generate the image conditioned on the semantic information.

For the IP-Adapter plus, we use a query network to get more information, it will learn more contents. 

If we use all the information of the reference image, I think it can learn to reconstruct this image, But that's not what we want

**Comment by danbochman** [comment]
@xiaohu2015 For my use-case I have to I need almost lossless querying of the image prompt
The adapter needs to guide fine-grained details such as buttons, text, small holes, etc.

The images still contain this information after resizing to 224x224 (For CLIP),
but do you think the CLIP encoder is able to capture these features?

Do you think this can be achieved by further modifications to the (Plus) Sampler or do I need a different encoder? For example, I was thinking about using the VAE's encoder...

**Comment by linxiang4200** [comment]
> > @xiaohu2015 thanks for your reply. I still have a bit of confusion about the first question. If the training image and the target image are the same, what is the model learning during the training process?感谢您的回复。对于第一个问题我还是有些困惑。如果训练图像和目标图像相同，那么训练过程中模型学习的是什么？
> 
> good question. 好问题。
> 
> For IP-Adapter, we use only global image embedding of CLIP image encoder (e.g. 1024 tensor for ViT-H), hence it only capture semantic information of the reference image, but can't reconstruct the original image, hence it learns to generate the image conditioned on the semantic information.对于IP-Adapter，我们仅使用CLIP图像编码器的全局图像嵌入（例如ViT-H的1024张量），因此它只能捕获参考图像的语义信息，但无法重建原始图像，因此它学习生成以语义信息为条件的图像。
> 
> For the IP-Adapter plus, we use a query network to get more information, it will learn more contents.对于IP-Adapter plus，我们使用查询网络来获取更多的信息，就会了解到更多的内容。
> 
> If we use all the information of the reference image, I think it can learn to reconstruct this image, But that's not what we want如果我们使用参考图像的所有信息，我认为它可以学习重建该图像，但这不是我们想要的

Okay, I understand. Thank you for your patient and detailed explanations. I really appreciate it.

**Comment by xiaohu2015** [comment]
> @xiaohu2015 For my use-case I need to I have almost lossless querying of the image prompt The adapter needs to guide fine-grained details such as buttons, text, small holes, etc.
> 
> The images still contain this information after resizing to 224x224 (For CLIP), but do you think the CLIP encoder is able to capture these features?
> 
> Do you think this can be achieved by further modifications to the (Plus) Sampler or do I need a different encoder? For example, I was thinking about using the VAE's encoder...

you can use all the tokens (356) from CLIP image encoder, I think it almost no information loss. Maybe you can also try other image encoder, e.g. DINOv2. But a big problem is that you may no longer be able to use text prompt ( the generation is dominated by image condition)

**Comment by George0726** [comment]
Thanks for your high-level explanation! I am curious about substituting the Image Encoder (CLIP) for multimodal encoders, like BLIPDiffusion, etc. Do you think it may extend the application scenarios of the current IP-Adapter?

**Comment by xiaohu2015** [comment]
> Thanks for your high-level explanation! I am curious about substituting the Image Encoder (CLIP) for multimodal encoders, like BLIPDiffusion, etc. Do you think it may extend the application scenarios of the current IP-Adapter?

yes, I think it will be many works for that like MiniGPT-5.
